numba_rvsdg/core/transformations.py

```diff
@@ -290,37 +291,99 @@ def find_tail_blocks(
     return tail_subregion
 
 
-def extract_region(scfg, region_blocks, region_kind):
+def extract_region(scfg: SCFG, region_blocks, region_kind, parent_region: RegionBlock):
     headers, entries = scfg.find_headers_and_entries(region_blocks)
     exiting_blocks, exit_blocks = scfg.find_exiting_and_exits(region_blocks)
     assert len(headers) == 1
     assert len(exiting_blocks) == 1
     region_header = next(iter(headers))
     region_exiting = next(iter(exiting_blocks))
 
+    # Generate a new region name
+    region_name = scfg.name_gen.new_region_name(region_kind)
     head_subgraph = SCFG(
         {name: scfg.graph[name] for name in region_blocks}, name_gen=scfg.name_gen
     )
 
-    if isinstance(scfg[region_exiting], RegionBlock):
-        region_exiting = scfg[region_exiting].exiting
-    else:
-        region_exiting = region_exiting
-
-    subregion = RegionBlock(
-        name=region_header,
+    def update_exiting(region_block: RegionBlock):
+        # Recursively updates the exiting blocks of a regionblock
+        region_exiting = region_block.exiting
+        region_exiting_block: BasicBlock = region_block.subregion.graph.pop(region_exiting)
+        jt = list(region_exiting_block._jump_targets)
```

Comment:
Why is there a conversion to `list` here. Can we not call `enumerate`  on the `_jump_targets` directly?